mm2src/coins/lp_coins.rs

```diff
@@ -1963,13 +1972,15 @@ pub trait TakerCoinSwapOpsV2: ParseCoinAssocTypes + CommonSwapOpsV2 + Send + Syn
         swap_unique_data: &[u8],
     ) -> Result<Self::Tx, TransactionErr>;
 
-    /// Wait until taker payment spend is found on-chain
-    async fn wait_for_taker_payment_spend(
+    /// Wait until taker payment spend transaction is found on-chain
+    async fn find_taker_payment_spend_tx(
         &self,
         taker_payment: &Self::Tx,
         from_block: u64,
         wait_until: u64,
-    ) -> MmResult<Self::Tx, WaitForPaymentSpendError>;
+    ) -> MmResult<Self::Tx, FindPaymentSpendError>;
+
+    async fn extract_secret_v2(&self, secret_hash: &[u8], spend_tx: &Self::Tx) -> Result<Vec<u8>, String>;
```

Comment:
Updated trait extract_secret_v2 https://github.com/KomodoPlatform/komodo-defi-framework/pull/2261/commits/90b80a3e60732a97a67e97e989593f5056458204

and decided it would be nice to have secret fixed array in args structures as well:
use maker_secret:[u8; 32] in SpendMakerPaymentArgs https://github.com/KomodoPlatform/komodo-defi-framework/pull/2261/commits/17b1c9405c0927704ae9444fe3982a954bc338f2
use taker_secret: [u8; 32] in RefundMakerPaymentSecretArgs https://github.com/KomodoPlatform/komodo-defi-framework/pull/2261/commits/2df4dd91bafe322566d0f0fa214f192d1e6e0874
use taker_secret: [u8; 32] in RefundFundingSecretArgs https://github.com/KomodoPlatform/komodo-defi-framework/pull/2261/commits/a2a448f58426b3b55b6d1f4eaa716679fafb6b00